src/io/recv_from.rs

```diff
@@ -31,22 +32,24 @@ impl<T: BoundedBufMut> Op<RecvFrom<T>> {
         msghdr.msg_name = socket_addr.as_ptr() as *mut libc::c_void;
         msghdr.msg_namelen = socket_addr.len();
 
-        Op::submit_with(
-            RecvFrom {
-                fd: fd.clone(),
-                buf,
-                io_slices,
-                socket_addr,
-                msghdr,
-            },
-            |recv_from| {
-                opcode::RecvMsg::new(
-                    types::Fd(recv_from.fd.raw_fd()),
-                    recv_from.msghdr.as_mut() as *mut _,
-                )
-                .build()
-            },
-        )
+        CONTEXT.with(|x| {
```

Comment:
I could see a macro serving well here, in all these op files, to reduce boilerplate and make future changes easier.